### PR TITLE
- reverted changes in a render method to fix a crash

### PR DESCRIPTION
**Pull request by AsterixxxGallier**

### CODE REVIEW

File: src/main/java/com/github/alexthe666/iceandfire/client/render/tile/RenderDreadPortal.java [hunk 1]
```diff
@@ -9,28 +9,28 @@
...
-
-    public void render(TileEntityDreadPortal te, double x, double y, double z, float partialTicks, int destroyStage, float alpha) {
+	private static final ResourceLocation END_SKY_TEXTURE = new ResourceLocation("iceandfire:textures/environment/dread_portal_background.png");
+	private static final ResourceLocation END_PORTAL_TEXTURE = new ResourceLocation("iceandfire:textures/environment/dread_portal.png");
+	private static final Random RANDOM = new Random(31100L);
+	private static final FloatBuffer MODELVIEW = GLAllocation.createDirectFloatBuffer(16);
+	private static final FloatBuffer PROJECTION = GLAllocation.createDirectFloatBuffer(16);
+	private final FloatBuffer buffer = GLAllocation.createDirectFloatBuffer(16);
+
+    public void render(@Nullable TileEntityDreadPortal te, double x, double y, double z, float partialTicks, int destroyStage, float alpha) {
```
Comment (SekretOne) [inline src/main/java/com/github/alexthe666/iceandfire/client/render/tile/RenderDreadPortal.java:25]: Nice- this seems to be the right annotation (for what this modpack has been using). Their using the javax annotations. This is a more explicit way to declare something can be nullable here.

File: src/main/java/com/github/alexthe666/iceandfire/client/render/tile/RenderDreadPortal.java [hunk 2]
```diff
@@ -9,28 +9,28 @@
...
         GlStateManager.disableLighting();
         OpenGlHelper.setLightmapTextureCoords(OpenGlHelper.lightmapTexUnit, 240.0F, 0.0F);
         RANDOM.setSeed(31100L);
         GlStateManager.pushMatrix();
         GlStateManager.getFloat(2982, MODELVIEW);
         GlStateManager.getFloat(2983, PROJECTION);
         double d0 = x * x + y * y + z * z;
-        int i = this.getPasses(d0);
+        int i = te == null ? 14 : this.getPasses(d0);
```
Comment (SekretOne) [inline src/main/java/com/github/alexthe666/iceandfire/client/render/tile/RenderDreadPortal.java:33]: So now your code cleanup helper won't misinterpret that this cannot be null.

**Review by SekretOne — APPROVED** [review]
Okay that should address the issue with this block. There's probably a less scattered way for us to render the item version- but not critical now.

**Review by SekretOne — COMMENTED** [review]
See some things

- [x] src/main/java/com/github/alexthe666/iceandfire/api/FoodUtils.java
- [x] src/main/java/com/github/alexthe666/iceandfire/block/BlockGoldPile.java
- [x] src/main/java/com/github/alexthe666/iceandfire/block/BlockSilverPile.java
    * item.getItem() looks like it's assuming it can't be null; is that safe?
- [x] src/main/java/com/github/alexthe666/iceandfire/client/gui/bestiary/GuiBestiary.java
    * item.getItem() looks like it's assuming it can't be null; is that safe?
- [x] src/main/java/com/github/alexthe666/iceandfire/client/model/ModelCyclops.java
    * public void setRotationAngles lines 394 to 409 used to be null wrapped on entity
- [x] src/main/java/com/github/alexthe666/iceandfire/client/render/tile/RenderJar.java
- [x] src/main/java/com/github/alexthe666/iceandfire/client/render/tile/RenderPixieHouse.java
- [x] src/main/java/com/github/alexthe666/iceandfire/compat/tinkers/TraitFreeze.java
    * `onHit` method
- [x] src/main/java/com/github/alexthe666/iceandfire/entity/ChainEntityProperties.java
    * line 79, entity may be nullable?
- [x] src/main/java/com/github/alexthe666/iceandfire/entity/EntityChainTie.java
     * line 56 doesn't do anything (and doesn't null check for next accessor)
- [x] src/main/java/com/github/alexthe666/iceandfire/entity/EntityDragonBase.java
    * lines 467-501,, can be massively simplified and null checked with:
    * ```java 
      Item item = stack.getItem();
      if( !stack.isEmpty() && item != null ){
          if(item == the.thingy()){
             ...
          }
        ```
    * `public void writeEntityToNBT(NBTTagCompound compound) {`  line 632 this.getCustomNameTag nullable
    * `public boolean processInteract(EntityPlayer player, EnumHand hand) {` has a lost null check
    * `public boolean isTargetBlocked(Vec3d target) {` same
- [x] src/main/java/com/github/alexthe666/iceandfire/entity/EntityDragonFireCharge.java
    * `public void onUpdate() ` possible null
    * `onUpdate()` a lot of null checks were removed
- [x] src/main/java/com/github/alexthe666/iceandfire/entity/EntityDragonIceCharge.java
- [x] src/main/java/com/github/alexthe666/iceandfire/entity/EntityDreadLichSkull.java
    * `protected void onHit(RayTraceResult raytraceResultIn)`
- [x] src/main/java/com/github/alexthe666/iceandfire/entity/EntityFireDragon.java
- [x] src/main/java/com/github/alexthe666/iceandfire/entity/EntityHippocampus.java
    * a fair number of null checks skipped- actually now that I'm looking closer, the forge ones like ItemStack are safe. Not sure about the saddle ones in here
- [ ] src/main/java/com/github/alexthe666/iceandfire/entity/EntityHippogryph.java
    * it's fully whitespace changes- going from 4 spaces to tab so it's hard to diff this massive file, maybe best to spin up mod and spawn one and see how it behaves to validate all fine.
- [x] essentially everything in the src/main/java/com/github/alexthe666/iceandfire/entity/ package

Now I don't think everything has to be put back- I think what's basically happened is:
    * Forge libs annotate anything actively as nullable, but don't conversely label *not nullable*,
    * Fire and Ice annotated things NotNullable ... but inconsistently. So the optimization worked by looking at the method, saw it not havingthe NotNullable, trusted and 'fixed', pushing the usage into nullity issues.

So, it thought the *contract was right* and the usage was wrong, not the other way around.

**Comment by SekretOne** [comment]
Another consideration: your code style doesn't match the base here. You're using tabs vs spaces. Not a 'problem', but you'll want to standardize one way or t'other.

**Comment by AsterixxxGallier** [comment]
fixed it all

**Pull request merged**